fence/agents/azure_arm/fence_azure_arm_network.py

```diff
@@ -0,0 +1,215 @@
+        try:
+            for vm in vms:
+                allOk = True
+
+                for nicRef in vm.network_profile.network_interfaces:
+                    logging.info("{get_nodes_list} testing network interface %s" % nicRef.id)
+
+                    nicresource = azure_fence_lib.get_azure_resource(nicRef.id)
+                    nic = network_client.network_interfaces.get(nicresource.ResourceGroupName, nicresource.ResourceName)
+                    if nic.ip_configurations:
+                        for ipConfig in nic.ip_configurations:
+                            logging.info("{get_nodes_list} testing network interface ip config")
+                            fenceSubnet = azure_fence_lib.get_fence_subnet_for_config(ipConfig, network_client)
+                            testOk = azure_fence_lib.test_fence_subnet(fenceSubnet, nic, network_client)
+                            if not testOk:
+                                allOk = False
+
+                if allOk:
+                    logging.info("{get_nodes_list} Virtual machine %s can be fenced" % vm.name)
+                    result[vm.name] = ("", None)
+                
+        except Exception as e:
+            fail_usage("{get_nodes_list} Failed: %s" % e)
+    else:
+        fail_usage("{get_nodes_list} No Azure clients configured. Contact support")
+
+    return result
+
+def get_power_status(clients, options):
+    logging.info("{get_power_status} getting power status for VM %s" % (options["--plug"]))
+    result = azure_fence_lib.FENCE_STATE_ON
+
+    if clients:
+        compute_client = clients[0]
+        network_client = clients[1]
+        rgName = options["--resourceGroup"]
+        vmName = options["--plug"]
+        
+        result = azure_fence_lib.get_power_status_impl(compute_client, network_client, rgName, vmName)
+    else:
+        fail_usage("{get_power_status} No Azure clients configured. Contact support")
+    
+    logging.info("{get_power_status} result is %s" % result)
+    return result
+
+def set_power_status(clients, options):
+    logging.info("{set_power_status} setting power status for VM " + options["--plug"] + " to " + options["--action"])
+
+    if clients:
+        compute_client = clients[0]
+        network_client = clients[1]
+        rgName = options["--resourceGroup"]
+        vmName = options["--plug"]
+        try:        
+            if (options["--action"]=="off"):
+                azure_fence_lib.set_power_status_off(compute_client, network_client, rgName, vmName)
+            elif (options["--action"]=="on"):
+                azure_fence_lib.set_power_status_on(compute_client, network_client, rgName, vmName)
+        except Exception as e:
+            fail_usage("Failed: %s" % e)
+    else:
+        fail_usage("No Azure clients configured. Contact support")
+
+    logging.info("{set_power_status} done")
+
+def define_new_opts():
+    all_opt["resourceGroup"] = {
```

Comment:
Same point as above about using instance metadata service to fetch this dynamically.